db/db_client/db_client_execute.go

```diff
@@ -144,10 +138,10 @@ func (c *DbClient) createTransaction(ctx context.Context, retryOnTimeout bool) (
 func (c *DbClient) startQuery(ctx context.Context, query string, tx *sql.Tx) (rows *sql.Rows, err error) {
 	doneChan := make(chan bool)
 	defer func() {
-		if err != nil && c.ensureSessionFunc != nil {
+		if err != nil {
 			// do this with a Background context, since the passed in context
```

Comment:
// if the underlying SQL client has certain errors (for example context expiry) it will reset the session
// restore the session data - prepared statements and introspection tables
// (do this with a Background context, since the passed in context may have expired)